src/azul/indexer/log_forwarding_service.py

```diff
@@ -28,7 +28,13 @@ class LogForwardingService(abc.ABC):
     def read_logs(self, bucket: str, key: str) -> Iterator[MutableJSON]:
         response = aws.s3.get_object(Bucket=bucket, Key=key)
         body = self._read_lines(response['Body'])
-        return self._read_logs(body)
+        for message in self._read_logs(body):
+            source_fields = {
+                '_source_bucket': bucket,
+                '_source_key': key,
+            }
+            message.update(source_fields)
```

Comment:
*If* you modify in place, why not use `message[…] = …`. It's two lines instead of five and avoids creating another dict. The only time I wouldn't do that if I have to add many entries.